### ISSUE TITLE
Fix `no-descending-specificity` false positives for modern CSS nesting

### ISSUE THREAD

**Issue by moeriki** [stylelint/stylelint#8849]
### What minimal example or steps are needed to reproduce the bug?

```css
.tng-button {
	.is-sm,
	&:where(:not(.is-md, .is-lg)) {
		.tng-icon {
			/*  */
		}
	}
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": { "no-descending-specificity": true }
}
```

### How did you run Stylelint?

`pnpm exec stylelint 'src/**/*.css' `

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint-config-standard",
  "stylelint-plugin-logical-css",
  "stylelint-config-recess-order"
}
```

### What did you expect to happen?

No problem reported.

### What actually happened?

```
✖  Expected selector ".tng-icon" to come before selector ".tng-button .is-sm .tng-icon"
```

**Comment by jeddy3** [comment]
@moeriki Thanks for the report and for using the template.

I can reproduce the issue [using the demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AXAdgcwLQCMCuKKEaABMADprkpICWAzlvQLYA0l1AZPAO4AWMAE4wAFPDQQUIuo2ZRWJGVgA2GAJRqyHFNVSYstSKQpUdZyMoiD4JYVADc26gF9trtO8ohWIAGa1lGAA5AENmOEQYAA8wgAdApDB6em9wYn8MBBATEhJyEEE8QPp8m2A8kAksWHowGDQoWn16WJgwWn92lABPUpIUQpgSd2dUowyAMStmEJQspHoewOUmgbAkACt6YlTYWJTEHIrF7uXVvvzlWZhF-PZSY6WYFbQULHHabEWQhpDBKAuICuKBuOhAlFGziAA).

The rule wants you to move `&:where(:not(.is-md, .is-lg))` before `.is-sm`, like so:

```css
.tng-button {	
	&:where(:not(.is-md, .is-lg)),
	.is-sm {
		.tng-icon {
			color: red;
		}
	}
}
```

Because it's resolving the original nesting to:

```css
.tng-button .is-sm .tng-icon {
  color: red;
}

.tng-button:where(:not(.is-md, .is-lg)) .tng-icon {
  color: red;
}
```

Where the first rule has a higher specificity (0.3.0) than the second one (0.2.0).

However, native CSS nesting and specificity [works differently](https://developer.mozilla.org/en-US/docs/Web/CSS/Guides/Nesting/Nesting_and_specificity).

I've created [an issue](https://github.com/stylelint/stylelint-config-recommended/issues/267) to remove it from the share config for now. It may become suitable again when we migrate to spec-based resolution using [`@csstools/selector-resolve-nested`](https://www.npmjs.com/package/@csstools/selector-resolve-nested/v/3.0.0).

**Comment by romainmenke** [comment]
This has been resolved since 17.0.0
